tests/instrumentation-framework/src/main/java/org/hawkular/apm/tests/dockerized/TestScenarioRunner.java

```diff
@@ -0,0 +1,230 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hawkular.apm.tests.dockerized;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import org.hawkular.apm.tests.dockerized.environment.DockerComposeExecutor;
+import org.hawkular.apm.tests.dockerized.environment.DockerImageExecutor;
+import org.hawkular.apm.tests.dockerized.environment.TestEnvironmentExecutor;
+import org.hawkular.apm.tests.dockerized.exception.TestFailException;
+import org.hawkular.apm.tests.dockerized.model.JsonPathVerify;
+import org.hawkular.apm.tests.dockerized.model.TestCase;
+import org.hawkular.apm.tests.dockerized.model.TestScenario;
+import org.hawkular.apm.tests.dockerized.model.Type;
+import org.hawkular.apm.tests.server.ApmMockServer;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.spotify.docker.client.exceptions.DockerCertificateException;
+
+/**
+ * Class for running test scenario.
+ *
+ * @author Pavol Loffay
+ */
+public class TestScenarioRunner {
+
+    /**
+     * Version of APM, needed for ENV variable inside docker image
+     */
+    private final String apmVersion;
+    /**
+     * Port at which will be listening APM server
+     */
+    private final int apmServerPort;
+
+    /**
+     * Mocked Hawkular APM server to collect information captured by agents.
+     */
+//    private final ApmMockServer apmServer;
+    private final ObjectMapper objectMapper;
+
+    private final ApmMockServer apmServer;
+
+    /**
+     * @param apmVersion - version of the Hawkular APM (usually current version)
+     * @param apmServerPort - port at which Hawkular APM server will be started
+     * @throws DockerCertificateException
+     */
+    public TestScenarioRunner(String apmVersion, int apmServerPort) throws DockerCertificateException {
+        this.apmVersion = apmVersion;
+        this.apmServerPort = apmServerPort;
+
+        this.objectMapper = new ObjectMapper();
+        apmServer = new ApmMockServer();
+        apmServer.setPort(apmServerPort);
+        apmServer.setShutdownTimer(60*60*1000);
+    }
+
+    /**
+     * Run a test scenario
+     *
+     * @param testScenario
+     * @return number of successful test cases
+     */
+    public int run(TestScenario testScenario) {
+        if (testScenario.getEnvironment().getDockerCompose() != null &&
+                testScenario.getEnvironment().getImage() != null) {
+            throw new IllegalArgumentException("Ambiguous environment: defined docker image and docker-compose");
+        }
+
+        System.out.println("Starting test scenario: " + testScenario);
```

Comment:
Yes, I think so. The important part is to be consistent with the code. I remember seeing some `log....` in one of the last files on this review.